app/models/claim/geckoboard_claim.rb

```diff
@@ -0,0 +1,71 @@
+require "geckoboard"
+
+class Claim
+  class GeckoboardClaim
+    attr_reader :claims
+
+    DATASET_FIELDS = [
+      Geckoboard::StringField.new(:reference, name: "Reference"),
+      Geckoboard::StringField.new(:policy, name: "Policy"),
+      Geckoboard::DateTimeField.new(:submitted_at, name: "Submitted at"),
+      Geckoboard::StringField.new(:checked, name: "Checked"),
+      Geckoboard::DateTimeField.new(:checked_at, name: "Checked at"),
+      Geckoboard::StringField.new(:check_result, name: "Check result"),
+      Geckoboard::StringField.new(:paid, name: "Paid"),
+      Geckoboard::DateTimeField.new(:paid_at, name: "Paid at"),
+    ]
+    BATCH_SIZE = 500
+
+    def initialize(claim_or_claims)
+      @claims = Array(claim_or_claims)
+    end
+
+    def save
+      batched_claims.each do |batch|
+        dataset.post(data_for_claims(batch))
+      end
+    end
+
+    private
+
+    def data_for_claims(claims)
+      claims.map do |claim|
+        {
+          reference: claim.reference,
+          policy: claim.policy.to_s,
+          submitted_at: claim.submitted_at,
+          checked: claim.check.present?.to_s,
+          checked_at: claim.check.present? ? claim.check.created_at : placeholder_date_for_nil_value,
+          check_result: claim.check.present? ? claim.check.result : "",
+          paid: claim.scheduled_payment_date.present?.to_s,
+          paid_at: claim.scheduled_payment_date.present? ? claim.scheduled_payment_date : placeholder_date_for_nil_value,
```

Comment:
This could be simplified to
```suggestion
          paid_at: claim.scheduled_payment_date.presence || placeholder_date_for_nil_value,
```